htdocs/expedition/list.php

```diff
@@ -246,11 +246,10 @@
 $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'user as u ON e.fk_user_author = u.rowid';
 
 // We'll need this table joined to the select in order to filter by sale
-if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) $sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
+if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc ON s.rowid = sc.fk_soc";
 if ($search_user > 0)
 {
-	$sql .= ", ".MAIN_DB_PREFIX."element_contact as ec";
-	$sql .= ", ".MAIN_DB_PREFIX."c_type_contact as tc";
+	$sql .= " INNER JOIN  ".MAIN_DB_PREFIX."element_contact as ec ON u.rowid = ec.element_id";
```

Comment:
The table u is the table to get name of the user fk_user_author.
But the search on user is to find object linked with element_id so we need a link between
ec.element_id and e.rowid